### PR TITLE
Only backup session if something has changed.

### LINKED ISSUE #1691: do not backup if there were no changes
even with no activity, backups happen
screen shot 2015-06-01 at 7 16 55 pm

### PR DESCRIPTION
**Pull request by samccone**
Fixes #1691

### CODE REVIEW

**Comment by scottdraves** [comment]
i don't like the idea of keeping around a whole extra copy of the notebook.
it would be better to either use a hash of it, or beaker is keeping track of the dirty state anyway, just use that.

**Comment by samccone** [comment]
~~found a bug in it one moment @scottdraves~~

resolved

**Comment by scottdraves** [comment]
This needs its own "edited" flag, not the one based on saving.
but that flag should be set by the same function that is called from many places in the app when the notebook is made dirty by editing.

**Comment by samccone** [comment]
ah I understand, sorry misstep on my part.

File: core/src/main/web/app/mainapp/services/sessionmanager.js [hunk 1]
```diff
@@ -832,7 +833,7 @@
         _format = format;
         _notebookUri.set(notebookUri);
         _notebookModel.set(notebookModel);
-        _edited = !!edited;
+        this.setNotebookModelEdited(!!edited);
```
Comment (ildipo) [inline core/src/main/web/app/mainapp/services/sessionmanager.js:836]: Semantic here is different - was this intentional?

File: core/src/main/web/app/mainapp/services/sessionmanager.js [hunk 2]
```diff
@@ -892,8 +893,11 @@
         }
       },
       backup: function() {
-        if (_sessionId && !_notebookModel.isEmpty()) {
-          return bkSession.backup(_sessionId, generateBackupData());
+        if (_sessionId && !_notebookModel.isEmpty() && _needsBackup) {
+          return bkSession.backup(_sessionId, generateBackupData())
+          .then(function() {
+            _needsBackup = false;
```
Comment (ildipo) [inline core/src/main/web/app/mainapp/services/sessionmanager.js]: Here I think you should clear _needsBackup before executing bkSession.backup() - what happens if I'm editing the notebook and I put edits during the time it takes for the request to go out to when the promise is fulfilled?

File: core/src/main/web/app/mainapp/services/sessionmanager.js (hunk 1 above)
Comment (samccone) [inline core/src/main/web/app/mainapp/services/sessionmanager.js:836]: can you expand on what you mean?

File: core/src/main/web/app/mainapp/services/sessionmanager.js (hunk 2 above)
Comment (samccone) [inline core/src/main/web/app/mainapp/services/sessionmanager.js]: k

**Pull request merged**